### DISCUSSION TITLE
WebREPL mode

### DISCUSSION THREAD

**Discussion by askvictor** [mu-editor/mu#1460] [Ideas & New Features]
I'm looking at restarting my (as my alias @eduvik) previous attempt at implementing a webREPL option for any devices that support it (currently ESP8266 and ESP32) for the new (>1.0) mode architecture.

In my previous work on this I used Qt libraries for websockets (and perhaps other things) though there was talk of removing the dependencies on Qt - what's the status of this at the moment? I had started on abstracting all communications (both serial and websocket) from the common functions using QtSerial and QtWebsockets, since the basic operations are the same regardless of the comms method - what are people's thoughts about that? The alternative is that the filemanager and repl code for webREPL will have a fair amount of duplicated code from the serial version.

**Comment by ZanderBrown** [comment]
I think at the point we've all accepted we are stuck with Qt fairly long term and mu itself uses QtSerial

**Reply by 4SiB** [reply]
And this QtSerial is not compatible with WebREPL anyway ?
This will be perfect to use WebREPL instead of serial communication.

+1 for WebREPL by any way :)
